tests/src/JIT/Directed/arglist/vararg.cs

```diff
@@ -0,0 +1,1195 @@
+                                                                                expectedValues[26],
+                                                                                expectedValues[27],
+                                                                                expectedValues[28],
+                                                                                expectedValues[29],
+                                                                                expectedValues[30],
+                                                                                expectedValues[31],
+                                                                                expectedValues[32],
+                                                                                expectedValues[33],
+                                                                                expectedValues[34],
+                                                                                expectedValues[35],
+                                                                                expectedValues[36],
+                                                                                expectedValues[37],
+                                                                                expectedValues[38],
+                                                                                expectedValues[39],
+                                                                                expectedValues[40]));
+
+        double sum = 0;
+        for (int i = 0; i < expectedValues.Length; ++i)
+        {
+            sum += expectedValues[i];
+        }
+
+        return sum == expectedSum;
+    }
+
+    /// <summary>
+    /// Given an input set create an arglist to pass to a vararg C function.
+    /// 
+    /// This is mostly to simulate calling printf from Managed code.
+    /// Based on the % values the c function will create and return a cstring
+    /// 
+    /// Compare the c string with the value expected, return true if they
+    /// are equal.
+    /// 
+    /// </summary>
+    /// <param name="expectedValues"></param>
+    /// <returns>bool</returns>
+    static bool TestPrintF()
+    {
+        bool success = true;
+
+        success = TestPrintFIntLongDouble("%d,%d,%f", 10, 11, 24.1) ? success : false;
+        success = TestPrintFDoubleDoubleIntIntLongDouble("%f,%f,%d,%d,%d,%f", 10.1, 11.1, 12, 2, 4L, 0.21) ? success : false;
+        success = TestPrintFManyLong() ? success : false;
+
+        return success;
+    }
+
+    /// <summary>
+    /// This is a helper for TestPrintF
+    ///
+    /// This will create a string with three arguments:
+    ///
+    /// int,
+    /// long,
+    /// double
+    /// 
+    /// </summary>
+    static bool TestPrintFIntLongDouble(string format, int val1, long val2, double val3)
+    {
+        printf(format + "\n", __arglist(val1, val2, val3));
```

Comment:
This just outputs to the console without testing the result? Is that useful, or was it just helpful to construct the test? (same question below)